### PR TITLE
Add SES-15 STIGs to exiting rules.

### PR DESCRIPTION
**Pull request by brett060102**
- Changes are to add SLES-15 STIG IDs and CCE ID to existing rules
- Change sle12 checks to include sle15
- Update sle 15 overlay file

- SLE-15 rules are:
    - SLES-15-020110 'accounts_have_homedir_login_defs'
    - SLES-15-040120 'accounts_user_home_paths_only'
    - SLES-15-040080 'accounts_user_interactive_home_directory_exists'
    - SLES-15-020020 'accounts_max_concurrent_login_sessions'
    - SLES-15-040110 'file_permission_user_init_files'
    - SLES-15-010130 'accounts_tmout'
    - SLES-15-040070 'accounts_user_interactive_home_directory_defined'
    - SLES-15-040420 'accounts_umask_etc_login_defs'
    - SLES-15-040130 'accounts_user_dot_no_world_writable_programs'
    - SLES-15-040090 'file_permissions_home_directories'
    - SLES-15-010260 'set_password_hashing_algorithm_logindefs'
    - SLES-15-020180 'set_password_hashing_min_rounds_logindefs'
    - SLES-15-020170 'set_password_hashing_algorithm_systemauth'
    - SLES-15-020080 'display_login_attempts'
    - SLES-15-040000 'accounts_passwords_pam_faildelay_delay'
    - SLES-15-020010 'accounts_passwords_pam_tally2'
    - SLES-15-020260 'cracklib_accounts_password_pam_minlen'
    - SLES-15-020150 'cracklib_accounts_password_pam_dcredit'
    - SLES-15-020270 'cracklib_accounts_password_pam_ocredit'
    - SLES-15-020290 'cracklib_accounts_password_pam_retry'
    - SLES-15-040220 'pam_disable_automatic_configuration'
    - SLES-15-010090 'dconf_gnome_login_banner_text'
    - SLES-15-010060 'banner_etc_gdm_banner'
    - SLES-15-010080 'dconf_gnome_banner_enabled'
    - SLES-15-010020 'banner_etc_issue'
    - SLES-15-020230 'accounts_password_set_max_life_existing'
    - SLES-15-020220 'accounts_maximum_age_login_defs'
    - SLES-15-020000 'account_temp_expire_date'
    - SLES-15-020060 'account_emergency_admin'
    - SLES-15-020050 'account_disable_post_pw_expiration'
    - SLES-15-020180 'accounts_password_all_shadowed_sha512'
    - SLES-15-020300 'no_empty_passwords'
    - SLES-15-010230 'account_unique_id'
    - SLES-15-010170 'smartcard_configure_ca'
    - SLES-15-010470 'smartcard_configure_cert_checking'
    - SLES-15-020030 'smartcard_pam_enabled'
    - SLES-15-010460 'install_smartcard_packages'
    - SLES-15-010110 'vlock_installed'
    - SLES-15-040060 'disable_ctrlaltdel_reboot'
    - SLES-15-010480 'kernel_module_usb-storage_disabled'
    - SLES-15-010240 'service_autofs_disabled'
    - SLES-15-010350 'file_permissions_var_log_messages'
    - SLES-15-030620 'permissions_local_audit_binaries'
    - SLES-15-030600 'permissions_local_var_log_audit'
    - SLES-15-010540 'sysctl_kernel_kptr_restrict'
    - SLES-15-010550 'sysctl_kernel_randomize_va_space'
    - SLES-15-040400 'no_files_unowned_by_user'
    - SLES-15-040180 'dir_perms_world_writable_system_owned_group'
    - SLES-15-010300 'dir_perms_world_writable_sticky_bits'
    - SLES-15-020240 'file_etc_security_opasswd'
    - SLES-15-040410 'file_permissions_ungroupowned'
    - SLES-15-040140 'mount_option_home_nosuid'
    - SLES-15-040150 'mount_option_nosuid_removable_partitions'
    - SLES-15-040360 'sysctl_net_ipv4_conf_default_send_redirects'
    - SLES-15-040370 'sysctl_net_ipv4_conf_all_send_redirects'
    - SLES-15-010310 'sysctl_net_ipv4_tcp_syncookies'
    - SLES-15-040330 'sysctl_net_ipv4_conf_all_accept_redirects'
    - SLES-15-040340 'sysctl_net_ipv4_conf_default_accept_redirects'
    - SLES-15-040320 'sysctl_net_ipv4_conf_default_accept_source_route'
    - SLES-15-040300 'sysctl_net_ipv4_conf_all_accept_source_route'
    - SLES-15-010380 'wireless_disable_interfaces'
    - SLES-15-010220 'kernel_module_dccp_disabled'
    - SLES-15-040310 'sysctl_net_ipv6_conf_all_accept_source_route'
    - SLES-15-040350 'sysctl_net_ipv6_conf_default_accept_source_route'
    - SLES-15-040390 'network_sniffer_disabled'
    - SLES-15-010220 'susefirewall2_only_required_services'
    - SLES-15-010220 'package_SuSEfirewall2_installed'
    - SLES-15-010220 'service_SuSEfirewall2_enabled'
    - SLES-15-040200 'partition_for_home'
    - SLES-15-040210 'partition_for_var'
    - SLES-15-030810 'partition_for_var_log_audit'
    - SLES-15-010450 'sudo_remove_no_authenticate'
    - SLES-15-010450 'sudo_remove_nopasswd'
    - SLES-15-020090 'accounts_authorized_local_users'
    - SLES-15-010000 'installed_OS_is_vendor_supported'
    - SLES-15-030630 'aide_check_audit_tools'
    - SLES-15-040050 'aide_verify_ext_attributes'
    - SLES-15-010420 'package_aide_installed'
    - SLES-15-040040 'aide_verify_acls'
    - SLES-15-010560 'clean_components_post_updating'
    - SLES-15-010010 'security_patches_up_to_date'
    - SLES-15-010430 'ensure_gpgcheck_globally_activated'
    - SLES-15-030050 'service_auditd_enabled'
    - SLES-15-030670 'package_audit-audispd-plugins_installed'
    - SLES-15-030650 'package_audit_installed'
    - SLES-15-030610 'audit_rules_unsuccessful_file_modification_truncate'
    - SLES-15-030370 'audit_rules_privileged_commands_ssh_agent'
    - SLES-15-030070 'audit_rules_privileged_commands_passwd'
    - SLES-15-030490 'audit_rules_privileged_commands_passmass'
    - SLES-15-030090 'audit_rules_privileged_commands_newgrp'
    - SLES-15-030560 'audit_rules_privileged_commands_sudo'
    - SLES-15-030340 'audit_rules_privileged_commands_chfn'
    - SLES-15-030550 'audit_rules_privileged_commands_su'
    - SLES-15-030120 'audit_rules_privileged_commands_chage'
    - SLES-15-030410 'audit_rules_privileged_commands_kmod'
    - SLES-15-030060 'audit_rules_privileged_commands_ssh_keysign'
    - SLES-15-030130 'audit_rules_privileged_commands_crontab'
    - SLES-15-030080 'audit_rules_privileged_commands_gpasswd'
    - SLES-15-030100 'audit_rules_privileged_commands_chsh'
    - SLES-15-030500 'audit_rules_privileged_commands_usermod'
    - SLES-15-030510 'audit_rules_privileged_commands_pam_timestamp_check'
    - SLES-15-030020 'audit_rules_usergroup_modification_shadow'
    - SLES-15-030000 'audit_rules_usergroup_modification_passwd'
    - SLES-15-030010 'audit_rules_usergroup_modification_group'
    - SLES-15-030640 'audit_rules_suid_privilege_function'
    - SLES-15-030470 'audit_rules_login_events_tallylog'
    - SLES-15-030480 'audit_rules_login_events_lastlog'
    - SLES-15-030040 'audit_rules_usergroup_modification_gshadow'
    - SLES-15-030440 'audit_rules_execution_chacl'
    - SLES-15-030430 'audit_rules_execution_setfacl'
    - SLES-15-030420 'audit_rules_execution_chmod'
    - SLES-15-030460 'audit_rules_execution_rm'
    - SLES-15-030420 'audit_rules_dac_modification_chmod'
    - SLES-15-030690 'auditd_audispd_configure_remote_server'
    - SLES-15-030590 'auditd_data_disk_full_action'
    - SLES-15-030680 'auditd_audispd_encrypt_sent_records'
    - SLES-15-030700 'auditd_data_retention_space_left'
    - SLES-15-030800 'auditd_audispd_disk_full_action'
    - SLES-15-030570 'auditd_data_retention_action_mail_acct'
    - SLES-15-010580 'rsyslog_remote_loghost'
    - SLES-15-010200 'grub2_uefi_password'
    - SLES-15-010190 'grub2_password'
    - SLES-15-010180 'package_telnet-server_removed'
    - SLES-15-040030 'no_host_based_files'
    - SLES-15-040020 'no_user_host_based_files'
    - SLES-15-040190 'service_kdump_disabled'
    - SLES-15-030580 'postfix_client_configure_mail_alias'
    - SLES-15-040240 'file_permissions_sshd_pub_key'
    - SLES-15-010150 'sshd_set_loglevel_verbose'
    - SLES-15-010280 'sshd_set_idle_timeout'
    - SLES-15-010160 'sshd_use_approved_ciphers'
    - SLES-15-020040 'sshd_disable_root_login'
    - SLES-15-040230 'sshd_disable_user_known_hosts'
    - SLES-15-010270 'sshd_use_approved_macs'
    - SLES-15-040270 'sshd_use_priv_separation'
    - SLES-15-040280 'sshd_disable_compression'
    - SLES-15-020120 'sshd_print_last_log'
    - SLES-15-040260 'sshd_enable_strictmodes'
    - SLES-15-010040 'sshd_enable_warning_banner'
    - SLES-15-010320 'sshd_set_keepalive_0'
    - SLES-15-040250 'file_permissions_sshd_private_key'
    - SLES-15-010530 'service_sshd_enabled'
    - SLES-15-040160 'mount_option_nosuid_remote_filesystems'
    - SLES-15-040170 'mount_option_noexec_remote_filesystems'
    - SLES-15-010400 'chronyd_or_ntpd_set_maxpoll'
    - SLES-15-010500 'sssd_offline_cred_expiration'
    - SLES-15-010490 'sssd_memcache_timeout'

#### Description:

- Changes are to add SLES-15 STIG IDs and CCE ID to existing rules
- Change sle12 checks to include sle15
- Update sle 15 overlay file

#### Rationale:

- Need to support SLE-15

### CODE REVIEW

File: linux_os/guide/system/software/updating/clean_components_post_updating/oval/sle15.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<def-group oval_version="5.10">
+  <definition class="compliance" id="clean_components_post_updating" version="1">
+    <metadata>
+      <title>Ensure Zypper Removes Previous Package Versions</title>
+      <affected family="unix">
+        <platform>SUSE Linux Enterprise 12</platform>
```
Comment (guangyee) [inline linux_os/guide/system/software/updating/clean_components_post_updating/oval/sle15.xml:6]: shouldn't this be 15?

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# platform = multi_platform_sle
```
Comment (guangyee) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml:1]: Really wish ssg.build_remediations.get_rule_dir_remediations() support the concept of product family so we don't have to copy these around.

File: linux_os/guide/system/software/updating/clean_components_post_updating/oval/sle15.xml (hunk 1 above)
Comment (brett060102) [reply linux_os/guide/system/software/updating/clean_components_post_updating/oval/sle15.xml]: yes, it should. Thanks and updated.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml (hunk 2 above)
Comment (brett060102) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml:1]: Me too. We could use links, but I never really liked using those in git.

**Comment by brett060102** [comment]
I see that the test error is:
39/71 Test #39: unique-cces .........................................................***Failed    0.35 sec

cce CCE-83055-4 is included in files: 
 - linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/rule.yml
 - linux_os/guide/system/network/network-uncommon/kernel_module_dccp_disabled/rule.yml
 
Looking at it now.

**Comment by brett060102** [comment]
Figure out the failure, surprised it did not trigger before. The revert of the during the SLe-12 work was not complete. A SLE-12 stig and CCE were left in the file and the CCE ID which was also in kernel_module_dccp_disabled caused the failure, I don't understand why adding kernel_module_dccp_disabled to the sl15 stig.profile would trigger the failure though.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml (hunk 2 above)
Comment (ggbecker) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/no_empty_passwords/ansible/sle15.yml:1]: Feel free to create a new issue asking for this. It seems reasonable.

**Comment by ggbecker** [comment]
> Figure out the failure, surprised it did not trigger before. The revert of the during the SLe-12 work was not complete. A SLE-12 stig and CCE were left in the file and the CCE ID which was also in kernel_module_dccp_disabled caused the failure, I don't understand why adding kernel_module_dccp_disabled to the sl15 stig.profile would trigger the failure though.

At first it was really strange to me as well... but I was able to understand why... the reason is because the following line contains trailing whitespaces and the test that checks for unique CCE is puzzled by that and considers a different CCE. So the [test](https://github.com/ComplianceAsCode/content/blob/master/tests/assert_reference_unique.sh) needs to be updated to discard any trailing whitespaces.

https://github.com/ComplianceAsCode/content/blob/3b7c65157b2defffb0bc89ee910725aa9fa5664d/linux_os/guide/system/network/network-uncommon/kernel_module_dccp_disabled/rule.yml#L22

File: linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/rule.yml [hunk 3]
```diff
@@ -41,6 +42,7 @@ references:
     srg: SRG-OS-000324-GPOS-00125,SRG-OS-000480-GPOS-00227
     stigid@rhel7: RHEL-07-020230
     stigid@sle12: SLES-12-010610
+    stigid@sle15: SLES-15-040060
     stigid@sle12: SLES-12-010611
```
Comment (ggbecker) [inline linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/rule.yml:46]: I think only one of these @sle12 identifiers will be considered

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_all_shadowed_sha512/rule.yml [hunk 4]
```diff
@@ -25,11 +25,13 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce@sle12: CCE-83038-0
+    cce@sle12: CCE-83038-0
```
Comment (ggbecker) [inline linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_all_shadowed_sha512/rule.yml:28]: trailing whitespaces added

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_chacl/rule.yml [hunk 5]
```diff
@@ -27,13 +27,15 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce@sle12: CCE-83190-9
+    cce@sle12: CCE-83190-9
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_chacl/rule.yml:30]: trailing whitespace

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_chacl/rule.yml [hunk 6]
```diff
@@ -27,13 +27,15 @@ rationale: |-
...
-    cce@sle12: CCE-83190-9
+    cce@sle12: CCE-83190-9 
+    cce@sle15: CCE-85595-7
 
 references:
     disa@sle12: CCI-000130,CCI-000169,CCI-000172,CCI-002884
     nist@sle12: AU-3,AU-3.1,AU-12.1(ii),AU-12(a),AU-12.1(iv),AU-12(c),MA-4(1)(a)
     srg@sle12: SRG-OS-000037-GPOS-00015,SRG-OS-000062-GPOS-00031,SRG-OS-000392-GPOS-00172,SRG-OS-000462-GPOS-00206,SRG-OS-000471-GPOS-00215
-    stigid@sle12: SLES-12-020620
+    stigid@sle12: SLES-12-020620
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_chacl/rule.yml:37]: trailing whitespace

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_setfacl/rule.yml [hunk 7]
```diff
@@ -27,12 +27,14 @@ rationale: |-
 severity: medium
 
 identifiers:
-    cce@sle12: CCE-83189-1
+    cce@sle12: CCE-83189-1
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_execution_acl_commands/audit_rules_execution_setfacl/rule.yml:30]: trailing whitespace

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_passmass/rule.yml [hunk 8]
```diff
@@ -31,12 +31,14 @@ severity: medium
...
 identifiers:
     cce@sle12: CCE-83193-3
+    cce@sle15: CCE-85599-9
 
 references:
     disa@sle12: CCI-000130,CCI-000169,CCI-000172,CCI-002884
     nist@sle12: AU-3,AU-12(a),AU-12(c),MA-4(1)(a)
     srg@sle12: SRG-OS-000037-GPOS-00015
-    stigid@sle12: SLES-12-020670
+    stigid@sle12: SLES-12-020670
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_passmass/rule.yml:40]: trailing whitespace

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml [hunk 9]
```diff
@@ -32,12 +32,14 @@ severity: medium
...
 identifiers:
     cce@sle12: CCE-83191-7
+    cce@sle15: CCE-85600-5
 
 references:
     disa@sle12: CCI-000130,CCI-000169,CCI-000172,CCI-002884
     nist@sle12: AU-3,AU-12(a),AU-12(c),MA-4(1)(a)
     srg@sle12: SRG-OS-000037-GPOS-00015
-    stigid@sle12: SLES-12-020700
+    stigid@sle12: SLES-12-020700
```
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml:41]: trailing whitespace

**Review by ggbecker — CHANGES_REQUESTED** [review]
Most of the changes are ok. Some minor requests. Please take a look

File: linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/rule.yml (hunk 3 above)
Comment (brett060102) [reply linux_os/guide/system/accounts/accounts-physical/disable_ctrlaltdel_reboot/rule.yml:46]: Yes, the original in the original disa SLES-12-010611 was a replete of SLES-12-010610, but SLES-12-010611 seems to be gone now. Will remove.

File: linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_all_shadowed_sha512/rule.yml (hunk 4 above)
Comment (brett060102) [reply linux_os/guide/system/accounts/accounts-restrictions/password_storage/accounts_password_all_shadowed_sha512/rule.yml:28]: Thanks, I though I caught those.

File: sle15/profiles/stig.profile [hunk 10]
```diff
@@ -14,5 +14,181 @@ description: |-
...
+    - sshd_disable_user_known_hosts
+    - sshd_enable_strictmodes
+    - sshd_enable_warning_banner
+    - sshd_print_last_log
+    - sshd_set_idle_timeout
+    - var_sshd_set_keepalive=0
+    - sshd_set_keepalive_0
+    - sshd_set_loglevel_verbose
+    - sshd_use_approved_ciphers
+    - sshd_use_approved_macs
```
[the remark this answers is not in the document]
Comment (brett060102) [reply sle15/profiles/stig.profile:175]: @ggbecker Thank you, and yes the 15 stigs want order, but not the sle-12 and the stig title for both are the same.

**Comment by brett060102** [comment]
Changes have been addressed, at least I see the last commit in https://github.com/brett060102/content/tree/add_sle15_rules but PR is slow to update, I did update 15 overlay for this change and that might be causing delay.

File: linux_os/guide/services/ssh/ssh_server/sshd_use_approved_ciphers/rule.yml [hunk 11]
```diff
@@ -49,7 +49,7 @@ severity: medium
 identifiers:
     cce@rhel7: CCE-27295-5
     cce@rhel8: CCE-81032-5
-    cce@sle12: CCE-83181-8
+    cce@sle12: CCE-83181-8
```
Comment (brett060102) [inline linux_os/guide/services/ssh/ssh_server/sshd_use_approved_ciphers/rule.yml:52]: I see the space

File: sle15/profiles/stig.profile [hunk 12]
```diff
@@ -14,5 +14,181 @@ description: |-
     DISA STIG for SUSE Linux Enterprise 15 V1R1.
 
 selections:
+    - sshd_approved_macs=stig
```
Comment (ggbecker) [inline sle15/profiles/stig.profile:17]: You can remove this variable since `sshd_use_approved_macs_ordered_stig` does not use it.

Comment (ggbecker) [inline sle15/profiles/stig.profile:17]:
```suggestion
```

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml (hunk 9 above)
Comment (ggbecker) [reply linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml:41]: this one still needs to be removed

File: sle15/profiles/stig.profile (hunk 12 above)
Comment (brett060102) [reply sle15/profiles/stig.profile]: Thank you and done.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml (hunk 9 above)
Comment (ggbecker) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_usermod/rule.yml:41]:
```suggestion
    stigid@sle12: SLES-12-020700
```

**Review by ggbecker — APPROVED** [review]
I've pushed or more commit (removing trailing whitespace), let's wait for CI results and this should be good to go.

**Comment by brett060102** [comment]
Thank you.

**Pull request merged**